### PR TITLE
Fix flaky integration test about "replicated max parallel fetches".

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Storages/StorageMerge.cpp [hunk 1]
```diff
@@ -676,14 +676,16 @@ void StorageMerge::convertingSourceStream(
         auto convert_actions_dag = ActionsDAG::makeConvertingActions(pipe.getHeader().getColumnsWithTypeAndName(),
                                                                      header.getColumnsWithTypeAndName(),
                                                                      ActionsDAG::MatchColumnsMode::Name);
-        auto actions = std::make_shared<ExpressionActions>(convert_actions_dag, ExpressionActionsSettings::fromContext(local_context, CompileExpressions::yes));
+        auto actions = std::make_shared<ExpressionActions>(
```
Comment (alexey-milovidov) [inline src/Storages/StorageMerge.cpp:679]: Unrelated change (whitespaces).

File: tests/integration/test_max_http_connections_for_replication/test.py [hunk 2]
```diff
@@ -114,5 +114,5 @@ def task(count):
     assert_eq_with_retry(node4, "select count() from test_table", "100")
     assert_eq_with_retry(node5, "select count() from test_table", "100")
 
-    # two per each host
-    assert node5.query("SELECT value FROM system.events where event='CreatedHTTPConnections'") == '4\n'
+    # Two per each host or sometimes less, if fetches are not performed in parallel. But not more.
+    assert node5.query("SELECT value FROM system.events where event='CreatedHTTPConnections'") <= '4\n'
```
Comment (alexey-milovidov) [inline tests/integration/test_max_http_connections_for_replication/test.py:118]: This is the fix.

Comment (alexey-milovidov) [reply tests/integration/test_max_http_connections_for_replication/test.py:118]: Less number of connections can be used legitimately.

**Review by FArthur-cmd — APPROVED** [review]
